pkg/registry/core/pod/rest/log.go

```diff
@@ -86,7 +88,7 @@ func (r *LogREST) Get(ctx context.Context, name string, opts runtime.Object) (ru
 
 	countSkipTLSMetric(logOpts.InsecureSkipTLSVerifyBackend)
 
-	if errs := validation.ValidatePodLogOptions(logOpts); len(errs) > 0 {
+	if errs := validation.ValidatePodLogOptions(logOpts, utilfeature.DefaultFeatureGate.Enabled(features.PodLogsQuerySplitStreams)); len(errs) > 0 {
```

Comment:
This block should probably be more like a normal API:

```
if !utilfeature.DefaultFeatureGate.Enabled(features.PodLogsQuerySplitStreams) {
    logOpts.Stream = nil
}
```

Then do validation as you are.  It feels duplicative, but validation needs to know if SOMEHOW the field was set when it should not be.